gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+    def test_check_python_packages(self, mock_convert_packages, mock_import, mock_version):
+        # Setup
+        class MockVersionSideEffect(object):
+            def __init__(self):
+                self.version = '3.0.2'
+
+        def version_side_effect(*args, **kwargs):
+            return MockVersionSideEffect()
+
+        mock_version.side_effect          = version_side_effect
+        mock_convert_packages.side_effect = lambda x: ['yaml', 'os']
+        mock_import.side_effect           = lambda x: None
+
+        # Default correct test
+        configuration_tests.check_python_packages('3.0.2', ['yaml', 'os'])
+        mock_convert_packages.assert_called_with(['yaml', 'os'])
+        mock_import.assert_called()
+        mock_version.assert_called_with('gslab_python')
+
+        # Incorrect gslab_python version
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_python_packages('3.0.3', ['yaml', 'os'])
+
+        # Module doesn't exist
+        def import_side_effect(*args, **kwargs):
+            if args[0] == 'os':
+                raise ImportError(args[0])
+
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            mock_import.side_effect = import_side_effect
+            configuration_tests.check_python_packages('3.0.2', ['yaml', 'os'])
+
+
+    def test_convert_packages_argument(self):
+        self.assertEqual(configuration_tests.convert_packages_argument(['test']), ['test'])
+        self.assertEqual(configuration_tests.convert_packages_argument(['test', 'test1']), 
+                                                                       ['test', 'test1'])
+        self.assertEqual(configuration_tests.convert_packages_argument('test'), ['test'])
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.convert_packages_argument(123)
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.convert_packages_argument({'test'})
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.convert_packages_argument(['test', 2])
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.convert_packages_argument([2, 2])
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.convert_packages_argument(lambda x: 'test')
+
+
+    @mock.patch('gslab_scons.configuration_tests.is_in_path')
+    @mock.patch('gslab_scons.configuration_tests.check_r_packages')
+    def test_check_r(self, mock_check_r_packages, mock_is_in_path):
+        # No R executable
+        mock_is_in_path.side_effect = lambda x: None
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_r()
+        mock_check_r_packages.assert_not_called()
+
+        # Test default arguments 
+        mock_is_in_path.side_effect = lambda x: True
```

Comment:
It could be a bit clearer to write `lambda x: (x == 'R')`, but I don't mind the current implementation.